tests/pypi/hub_builder/hub_builder_tests.bzl

```diff
@@ -496,34 +632,34 @@ torch==2.4.1+cpu ; platform_machine == 'x86_64' \
     pypi.group_map().contains_exactly({})
     pypi.whl_map().contains_exactly({
         "torch": {
-            "pypi_312_torch_cp312_cp312_linux_x86_64_8800deef": [
+            "pypi_312_torch_cp312_cp312_linux_x86_64_8800deef_linux_x86_64": [
                 whl_config_setting(
-                    target_platforms = ("cp312_linux_x86_64",),
+                    target_platforms = ["cp312_linux_x86_64"],
                     version = "3.12",
                 ),
             ],
-            "pypi_312_torch_cp312_cp312_manylinux_2_17_aarch64_36109432": [
+            "pypi_312_torch_cp312_cp312_manylinux_2_17_aarch64_36109432_linux_aarch64": [
                 whl_config_setting(
-                    target_platforms = ("cp312_linux_aarch64",),
+                    target_platforms = ["cp312_linux_aarch64"],
                     version = "3.12",
                 ),
             ],
-            "pypi_312_torch_cp312_cp312_win_amd64_3a570e5c": [
+            "pypi_312_torch_cp312_cp312_win_amd64_3a570e5c_windows_x86_64": [
```

Comment:
This is not ideal that we are adding extra suffixes here, but I would need to refactor the code a lot to do various book keeping otherwise. TBH, we could potentially just use the target platform here instead of the whl specification or choose the shorter of the 2 variants:
- `<prefix>_<sanitized whl_name>`
- `<prefix>_<distro_name>_<suffixes>`